src/Stormpath/Authc/UsernamePasswordRequest.php

```diff
@@ -22,12 +22,15 @@ class UsernamePasswordRequest implements AuthenticationRequest
     private $username;
     private $password;
     private $host;
+    private $accountStore;
 
-    public function __construct($username, $password, $host = null)
+    public function __construct($username, $password, array $options = array())
     {
-        $this->host = $host;
         $this->password = $password ? str_split($password) : array();
         $this->username = $username;
+
+        $this->host = isset($options['host']) ? $options['host'] : null;
+        $this->accountStore = isset($options['accountStore']) ? $options['accountStore'] : null;
```

Comment:
It would be more like an assertion that the accountStore is an instance of AccountStore, instead of ignoring the update if it isn't. Probably throwing an `IllegalArgumentException` with a message if it is not an instance of that interface to let the user know what happened (otherwise the user might think that the request went through the way it was supposed to).